packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/editingcontext/services/EditingContextDispatcher.java

```diff
@@ -78,16 +90,25 @@ public Mono<IPayload> dispatchQuery(String editingContextId, IInput input) {
 
     @Override
     public Mono<IPayload> dispatchMutation(String editingContextId, IInput input) {
-        var canView = false;
-        var optionalProjectId = this.projectEditingContextService.getProjectId(editingContextId);
-        if (optionalProjectId.isPresent()) {
-            canView = this.capabilityEvaluator.hasCapability(SiriusWebCapabilities.PROJECT, optionalProjectId.get(), SiriusWebCapabilities.Project.EDIT);
+        var canEdit = false;
+
+        if (input instanceof IFormInput formInput) {
+            canEdit = this.editingContextSearchService.findById(editingContextId)
+                    .map(editingContext -> this.formCapabilitiesService.getFormCapabilities(formInput.representationId(), editingContext))
+                    .map(FormCapabilitiesDTO::canEdit)
+                    .orElse(false);
```

Comment:
You cannot reload the entire editing context from scratch here anytime there is a new input received